includes/class-wc-payments-order-success-page.php

```diff
@@ -16,60 +16,45 @@ class WC_Payments_Order_Success_Page {
 	 * Constructor.
 	 */
 	public function __construct() {
-		add_filter( 'woocommerce_thankyou_order_received_text', [ $this, 'show_woopay_thankyou_notice' ], 10, 2 );
-		add_action( 'woocommerce_thankyou_woocommerce_payments', [ $this, 'show_woopay_payment_method_name' ] );
+		add_filter( 'woocommerce_order_get_payment_method_title', [ $this, 'show_woopay_payment_method_name' ], 10, 2 );
 		add_filter( 'woocommerce_thankyou_order_received_text', [ $this, 'add_notice_previous_paid_order' ], 11 );
 		add_filter( 'woocommerce_thankyou_order_received_text', [ $this, 'add_notice_previous_successful_intent' ], 11 );
 		add_action( 'wp_enqueue_scripts', [ $this, 'enqueue_scripts' ] );
 	}
 
-
 	/**
-	 * Return the WooPay thank you notice when the order was created via WooPay
-	 *
-	 * @param string        $text  the default thank you text.
-	 * @param WC_Order|null $order the order being shown.
+	 * Add the WooPay logo and the last 4 digits of the card used to the payment method name
+	 * on the order received page.
 	 *
-	 * @return string
+	 * @param string            $payment_method_title the default payment method title.
+	 * @param WC_Abstract_Order $abstract_order the order being shown.
 	 */
-	public function show_woopay_thankyou_notice( $text, $order ) {
-		if ( ! $order || ! $order->get_meta( 'is_woopay' ) ) {
-			return $text;
-		}
-
-		return '<div class="thankyou-notice-woopay">' . __( 'Thank you! We’ve received your order.', 'woocommerce-payments' ) . '</div>';
-	}
+	public function show_woopay_payment_method_name( $payment_method_title, $abstract_order ) {
 
+		// Only change the payment method title on the order received page.
+		if ( ! is_order_received_page() ) {
+			return $payment_method_title;
+		}
 
-	/**
-	 * Add a WooPay logo and card last 4 to the payment method name in the success page.
-	 *
-	 * @param int $order_id the order id.
-	 */
-	public function show_woopay_payment_method_name( $order_id ) {
-		$order = wc_get_order( $order_id );
+		$order_id = $abstract_order->get_id();
+		$order    = wc_get_order( $order_id );
 		if ( ! $order || ! $order->get_meta( 'is_woopay' ) ) {
-			return;
+			return $payment_method_title;
 		}
-		?>
-		<ul class="woocommerce-order-overview woocommerce-thankyou-order-details order_details woopay">
```

Comment:
@rafaelzaleski since you are removing this element, I think you should also remove the CSS definitions introduced in #4232 for this element. Especifically the ones using the `.woopay` class since it's gone now.
- `.woocommerce-order-overview.woocommerce-thankyou-order-details.order_details.woopay`
- `.woocommerce-order-overview.woocommerce-thankyou-order-details.order_details.woopay
	.woocommerce-order-overview__payment-method.method` 

Which are still present in the `assets/css/success.css` file.

Reply:
Good catch! Removed on https://github.com/Automattic/woocommerce-payments/pull/6740/commits/a64bf174e66f0fba53274e7c9857858f53ec2318